src/Integrations/POSCatalog/ProductMapper.php

```diff
@@ -20,6 +20,31 @@
  * WIP class, copied from the `mobile/pos-catalog` branch.
  */
 class ProductMapper implements ProductMapperInterface {
+	/**
+	 * Fields to include in the product mapping.
+	 *
+	 * @var array|null Fields to include in the product mapping.
+	 */
+	private $fields = null;
+
+	/**
+	 * Set fields to include in the product mapping.
+	 *
+	 * @param string|null $fields Fields to include in the product mapping.
+	 * @return void
+	 */
+	public function set_fields( ?string $fields = null ): void {
```

Comment:
I think we can always force $fields as string. Not sure why we need to make it nullable. Then we can remove the redundant null handling. 

```suggestion
	public function set_fields( string $fields = null ): void {
```

Reply:
It's mostly there for testing purposes. At the moment, it doesn't matter, but if we add multiple tests, we should be able to unset it.

https://github.com/woocommerce/woocommerce-product-feed-for-openai/blob/ebd042138d5aced5a4d5ecb875679f86e63ba8e3/tests/unit/ProductFeed/Integrations/POSCatalog/ProductMapperTest.php#L26-L30

Reply:
😬 I would prefer a kind of dangerous name to set this to be null, something like `public function __set_null_fields_for_test()`. Or... use Reflection to achieve that. 

But anyway, not a blocker for this issue.

Reply:
It's just a nullable field, and calling it with `null` will simply remove the filter. It's nothing dangerous.